### PR TITLE
[:pineapple: < MG] operator<< implementations for a few foundational types

### PR DESCRIPTION
**Pull request by spacejam**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Update [changelog](https://docs.memgraph.com/memgraph/changelog)
- [ ] Provide the full content or a guide for the final git message

### CODE REVIEW

File: src/storage/v3/schemas.hpp [hunk 1]
```diff
@@ -25,11 +25,22 @@
...
+std::string SchemaTypeToString(common::SchemaType type);
+
 struct SchemaProperty {
   PropertyId property_id;
   common::SchemaType type;
 
   friend bool operator==(const SchemaProperty &lhs, const SchemaProperty &rhs);
+
+  friend std::ostream &operator<<(std::ostream &in, const SchemaProperty &schema_property) {
+    in << "SchemaProperty { property_id: ", in << (int)schema_property.property_id.AsUint();
```
Comment (antaljanosbenjamin) [inline src/storage/v3/schemas.hpp:37]:
```suggestion
    in << "SchemaProperty { property_id: ", in << schema_property.property_id.AsUint();
```

This is "C-style" cast, which is a bad practice in C++. If you need to cast the `uint64_t` number to an integer (for the first glance I don't see why it is necessary, but I might be wrong), then use `static_cast<int>(schema_property.property_id.AsUint())` instead, or if `int64_t` is enough, then `schema_property.property_id.AsInt()`.

File: src/coordinator/hybrid_logical_clock.hpp [hunk 2]
```diff
@@ -31,6 +34,15 @@ struct Hlc {
   bool operator==(const uint64_t other) const { return logical_id == other; }
   bool operator<(const uint64_t other) const { return logical_id < other; }
   bool operator>=(const uint64_t other) const { return logical_id >= other; }
+
+  friend std::ostream &operator<<(std::ostream &in, const Hlc &hlc) {
+    auto wall_clock = std::chrono::system_clock::to_time_t(hlc.coordinator_wall_clock);
+    in << "Hlc { logical_id: " << (int)hlc.logical_id;
```
Comment (antaljanosbenjamin) [inline src/coordinator/hybrid_logical_clock.hpp:40]: I started to think I am missing something here with the casts to `int`, so please enlighten me!

File: src/storage/v3/id_types.hpp [hunk 3]
```diff
@@ -35,6 +36,11 @@ namespace memgraph::storage::v3 {
     constexpr uint64_t AsUint() const { return id_; }                                                                 \
     constexpr int64_t AsInt() const { return std::bit_cast<int64_t>(id_); }                                           \
                                                                                                                       \
+    friend std::ostream &operator<<(std::ostream &in, const name &n) {                                                \
+      in << (int)n.id_;                                                                                               \
```
Comment (antaljanosbenjamin) [inline src/storage/v3/id_types.hpp:40]: Maybe use `AsInt` if the type is important.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]

**Review by gvolfing — APPROVED** [review]

File: src/storage/v3/schemas.hpp (hunk 1 above)
Comment (spacejam) [reply src/storage/v3/schemas.hpp:37]: after testing, I've confirmed what you said about it only being weird for byte-sized types, so I'll remove all casting here!

File: src/storage/v3/id_types.hpp (hunk 3 above)
Comment (spacejam) [reply src/storage/v3/id_types.hpp]: added, thanks

**Review by antaljanosbenjamin — APPROVED** [review]

File: src/storage/v3/id_types.hpp [hunk 4]
```diff
@@ -35,6 +36,11 @@ namespace memgraph::storage::v3 {
     constexpr uint64_t AsUint() const { return id_; }                                                                 \
     constexpr int64_t AsInt() const { return std::bit_cast<int64_t>(id_); }                                           \
                                                                                                                       \
+    friend std::ostream &operator<<(std::ostream &in, const name &n) {                                                \
+      in << n.id_.AsInt();                                                                                            \
```
Comment (antaljanosbenjamin) [inline src/storage/v3/id_types.hpp:40]:
```suggestion
      in << AsInt();                                                                                            \
```

`AsInt` is a member function of this class, not the `uint64_t` (as primitive types doesn't have any member functions).

**Pull request merged**